lib/combine_popolo_memberships.rb

```diff
@@ -1,30 +1,50 @@
 require 'combine_popolo_memberships/version'
+require 'date-range'
 
 # Takes multiple popolo membership arrays and combines them based on date.
 module CombinePopoloMemberships
-  def self.overlap(mem, term)
-    mS = mem[:start_date].to_s.empty? ? '0000-00-00' : mem[:start_date].to_s
-    mE = mem[:end_date].to_s.empty? ? '9999-99-99' : mem[:end_date].to_s
-    tS = term[:start_date].to_s.empty? ? '0000-00-00' : term[:start_date].to_s
-    tE = term[:end_date].to_s.empty? ? '9999-99-99' : term[:end_date].to_s
-
-    return unless mS < tE && mE > tS
-    (s, e) = [mS, mE, tS, tE].sort[1, 2]
-    {
-      _data: [mem, term],
-      start_date: s == '0000-00-00' ? nil : s,
-      end_date:   e == '9999-99-99' ? nil : e
-    }
+  class Membership
+    def initialize(raw)
+      @raw = raw
+    end
+
+    def start_date
```

Comment:
Good idea. I've switched to getting the dates through `date_range` in the tests in bfc873d and then made `start_date` and `end_date` private in 4dd049d.